controller/app/controllers/otp/validator/OtpRequestValidator.java

```diff
@@ -102,4 +102,39 @@ private void validateType(String type) {
                   JsonKey.RECOVERY_PHONE)));
     }
   }
+
+  public void validateGenerateOtpRequestV3(Request otpRequest) {
+    commonValidationV3(otpRequest, false);
+    validateTemplateId(otpRequest);
+  }
+
+
+  private void commonValidationV3(Request otpRequest, boolean isOtpMandatory) {

```

Comment:
This method suppose to be 
validateParam(
            (String) otpRequest.getRequest().get(JsonKey.CONTEXT_TYPE),
            ResponseCode.mandatoryParamsMissing,
            JsonKey.KEY);
 commonValidation(otpRequest, false);
    validateTemplateId(otpRequest);

Reply:
Refractorded the methods.